src/API.jl

```diff
@@ -892,21 +892,28 @@ function build(ctx::Context, pkgs::Vector{PackageSpec}; verbose=false, kwargs...
     Operations.build(ctx, pkgs, verbose)
 end
 
-precompile() = precompile(Context())
-function precompile(ctx::Context)
+precompile(;kwargs...) = precompile(Context(); kwargs...)
+function precompile(ctx::Context; parallel=true)
     printpkgstyle(ctx, :Precompiling, "project...")
     
-    num_tasks = parse(Int, get(ENV, "JULIA_NUM_PRECOMPILE_TASKS", string(Sys.CPU_THREADS + 1)))
+    num_tasks = parallel ? parse(Int, get(ENV, "JULIA_NUM_PRECOMPILE_TASKS", string(Sys.CPU_THREADS + 1))) : 1
     parallel_limiter = Base.Semaphore(num_tasks)
     
-    man = Pkg.Types.read_manifest(ctx.env.manifest_file)
-    pkgids = [Base.PkgId(first(dep), last(dep).name) for dep in man if !Pkg.Operations.is_stdlib(first(dep))]
-    pkg_dep_uuid_lists = [collect(values(last(dep).deps)) for dep in man if !Pkg.Operations.is_stdlib(first(dep))]
-    filter!.(!is_stdlib, pkg_dep_uuid_lists)
+    toplevel_pkgids = [Base.PkgId(uuid, name) for (name, uuid) in ctx.env.project.deps if !is_stdlib(uuid)]
+    
+    if parallel
+        man = Pkg.Types.read_manifest(ctx.env.manifest_file)
+        pkgids = [Base.PkgId(first(dep), last(dep).name) for dep in man if !Pkg.Operations.is_stdlib(first(dep))]
+        pkg_dep_uuid_lists = [collect(values(last(dep).deps)) for dep in man if !Pkg.Operations.is_stdlib(first(dep))]
+        filter!.(!is_stdlib, pkg_dep_uuid_lists)
+    else
+        pkgids = toplevel_pkgids
+        pkg_dep_uuid_lists = fill(Base.UUID[], length(pkgids)) # give empty arrays because we're not waiting for deps
```

Comment:
Wouldn't this trick break when a top-level package depends on other top-level packages?

If we are to do this, I think an easier option is to bring back the old `precompile` as a separate function and call it when `parallel == false`.  If we reuse the `is_stale` function, I guess it's not a very big function?

Reply:
Ah, yes, you're right. I also wondered whether the `parallel=false` functionality was even needed now, with allowing failures for sub-depepdencies (is that a good name for them??)?
The user can always just precompile the classical way via code-load

Reply:
> whether the `parallel=false` functionality was even needed now

This is exactly what I'm just wondering now, too.  We have `JULIA_NUM_PRECOMPILE_TASKS` to control the number of tasks so I think it'd be better to not implement this right now.

Reply:
Ok, I'll remove it. It can always be added back if needed